core/src/main/java/ai/timefold/solver/core/impl/solver/termination/DiminishedReturnsTermination.java

```diff
@@ -97,6 +98,7 @@ public void step(long time, InnerScore<Score_> bestScore) {
     private void resetGracePeriod(long currentTime, InnerScore<Score_> startingScore) {
         gracePeriodStartTimeNanos = currentTime;
         isGracePeriodActive = true;
+        locked = false;
```

Comment:
Why can't we not `isGracePeriodActive` for this? Seems to be the same thing.

And if we can't, this IMO needs a better name; Like "isGracePeriodActive". :-)

Reply:
The `locked` variable will be updated only once when the grace period is activated for the first time.

There are two evaluation phases: one during the grace period and another after it ends. If we begin using the activation variable, we will not trigger the second evaluation, which occurs outside the grace period evaluation block.

Returning `false` for `!isGracePeriodActivated` would cause us to skip the second termination evaluation phase, creating an inconsistency.